app/lib/features/news/model/news_references_model.dart

```diff
@@ -1,13 +1,30 @@
 enum NewsReferencesType {
-  shareEvent,
+  calendarEvent,
+  link;
+
+  static NewsReferencesType? fromStr(String typeStr) {
+    return values.asNameMap()[_toCamelCase(typeStr)];
+  }
+}
+
+String _toCamelCase(String s) {
+  final words = s
+      .split('-')
+      .map((w) =>
+          '${w.substring(0, 1).toUpperCase()}${w.substring(1).toLowerCase()}',)
+      .toList();
+  words[0] = words[0].toLowerCase();
+  return words.join();
```

Comment:
Not quite understand this code.

Reply:
it rewrites strings from `things-named-like-this` to `thingsNamedLikeThis`